### DISCUSSION TITLE
Server side pagination with V8: `getCanNextPage` issue.

### DISCUSSION THREAD

**Discussion by colindjk** [TanStack/table#3822] [General] [answered]
Hello again, loving V8 right now. One thing I noticed though while doing SS pagination is that the `getCanNextPage` only uses the row count of the client side rows, meaning if you're attempting to handle pagination server side, this function will likely return false.

Instead I think it should use the `pageCount` if the `pageCount` is provided by the implementer. 

File: `packages/table-core/src/features/Pagination.ts`
```tsx
      getCanNextPage: () => {
        const { pageIndex, pageSize } = instance.getState().pagination

        const pageCount = instance.getPageCount()

        if (pageCount === -1) {
          return true
        }

        if (pageCount === 0) {
          return false
        }

        // Instead of calculating here, just rely on pageCount.
        return (
          pageIndex < pageCount - 1
        )
      },
```

I can create a PR for this if you think it makes sense to change, would love to contribute to one of your repos.

**Comment by colindjk** [comment]
@tannerlinsley curious, is this implementation intentional? If not I can open up a PR. 

Function in question - https://github.com/TanStack/react-table/blob/bd8e89abb35be7c1a4daf51fa2c0b97f3b8c8737/packages/table-core/src/features/Pagination.ts#L189

**Comment by tannerlinsley** [comment, marked as answer]
Happy to review a PR to make it better.

Tanner Linsley

**Reply by colindjk** [reply]
Sounds good, opened up a PR here - https://github.com/TanStack/react-table/pull/3829

**Comment by talatkuyuk** [comment]
Hi @colindjk, I am trying to implement **manual pagination with react table v.8**. The example in the docs is related with the client-side pagination. I suppose you have implemented a server-side pagination as far as I understand from your PR. Can you share the implementation in order to figure out and inspire how to use manual pagination in v.8?

**Comment by tannerlinsley** [comment]
Yeah sure!

Tanner Linsley
